backend: initial tags api support (addresses #365):

ikreymer opened the pull request:
- add 'tags' field to crawlconfig (array of strings)
- allow querying crawlconfigs to specify multiple 'tag' query args, eg. tag=A&tag=B
- tags list accessible via `/archives/{aid}/crawlconfigs/tags`
- tests: add tests for tag creation and filtering, also fix potential errors with test fixtures by adding retry

ikreymer committed:
backend: initial tags api support (addresses #365):
- add 'tags' field to crawlconfig (array of strings)
- allow querying crawlconfigs to specify multiple 'tag' query args, eg. tag=A&tag=B

SuaYoo commented:
@ikreymer could you deploy this one to dev so I can test with https://github.com/webrecorder/browsertrix-cloud/pull/422?

ikreymer committed:
add tags distinct

ikreymer committed:
tags work

ikreymer committed:
index: update index to include archive id
tests: add tests for filtering by tags

ikreymer committed:
fix typos

ikreymer committed:
fix index

ikreymer committed:
tests: cleanup unused
tests: run crawl seed: use https://webrecorder.net/ instead of https://example.com/

ikreymer committed:
tests: add wait for auth login

ikreymer committed:
fix typo

tw4l commented on backend/test/conftest.py:
```diff
@@ -14,24 +15,33 @@
+        )
+        data = r.json()
+        try:
+            return {"Authorization": f"Bearer {data['access_token']}"}
```
nice! thanks for this :)

tw4l commented on backend/btrixcloud/crawlconfigs.py:
```diff
@@ -359,12 +365,17 @@ async def update_crawl_config(self, cid: uuid.UUID, update: UpdateCrawlConfig):
         return {"success": True}
 
-    async def get_crawl_configs(self, archive: Archive):
+    async def get_crawl_configs(self, archive: Archive, tags: List[str] = None):
```
should this be `..., tags: Union[List[str], None] = None`? or `Optional[List[str]] = None` I suppose

tw4l reviewed (approved):
One typing question but otherwise LGTM!

ikreymer committed:
types: mark as Optional

ikreymer committed:
reformat
